### DISCUSSION TITLE
How to generate CUID in migrations?

### DISCUSSION THREAD

**Discussion by villesau** [prisma/prisma#17743] [Q&A]
I need to generate new records in the migration. However the migration files are SQL files, and the CUID seems  to be generated by Prisma client I'm not able to find a way to use CUID. How can this be achieved? The idea is to split one table in two tables and I would not want to reuse the ids.

**Comment by arosemena** [comment]
You can import the cuid package manually into the migration and generate them that way:

https://github.com/paralleldrive/cuid

Create an empty migration and write the code that splits the table there, whenever you need a new cuid just call the `cuid()` function.

**Reply by ludralph** [reply]
Hi @villesau 

Do you have any more questions around this issue? If no, it would be really helpful if you could quickly **mark this Discussion as _answered_** to indicate that your issue has been resolved. 

In case you do have any outstanding questions, feel free to let us know and we'd be happy to help!

**Reply by villesau** [reply]
Unfortunately that does not answer to my question. Prisma only supports SQL migrations and the package linked is javascript..

**Reply by alan345** [reply]
Prisma is using cuid1 or 2?

https://github.com/paralleldrive/cuid2

**Comment by madhugb** [comment]
@villesau Its probably too late! but I was able to do this from MySQL migrations. 
Documented the process here - https://gist.github.com/madhugb/6ab4a341e3f3d67b6355d3f954e4400d

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
